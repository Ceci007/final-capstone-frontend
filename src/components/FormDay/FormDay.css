@import "https://fonts.googleapis.com/css2?family=Roboto&display=swap";

.day {
  font-family: 'Roboto', sans-serif;
  display: flex;
  justify-content: flex-end;
  border: 1px solid gray;
  background-color: white;
  margin: 0 15px 12px 15px;
  border-radius: 11px;
  flex-direction: column;
  font-size: 3.5vw;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.day input {
  border-radius: 5px;
  border: 1px solid gray;
  font-size: 3.2vw;
}

.date-temp {
  display: flex;
  margin: 8px 0 8px 7px;
  justify-content: center;
}

.date-div,
.temp-div {
  display: flex;
  align-items: center;
}

.date-div {
  margin-right: 8px;
}

.date-div input,
.temp-div input {
  width: 32vw;
  font-size: 3.2vw;
  font-family: 'Roboto', sans-serif;
  padding: 6px;
}

.temp-div input {
  width: 9vw;
}

.mood-div {
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  align-items: center;
}

.mood-div span {
  font-size: 5vw;
}

.mood-div span:first-child {
  font-size: 4vw;
}

.medicines2,
.symptons2 {
  display: flex;
  justify-content: center;
  margin: 0;
  padding: 0 10px;
  list-style: none;
}

.medicines2 p,
.symptons2 p {
  width: 22vw;
  text-align: center;
}

.medicine2 div,
.symptons2 div {
  display: flex;
  flex-direction: column;
}

.med-div {
  width: 50%;
}

.med-div input,
.symp-div input {
  margin: 0.3vh 0;
  width: 40vw;
  padding: 6px;
}

.med-div input::placeholder,
.symp-div input::placeholder {
  padding-left: 5px;
}

.day-buttons {
  justify-content: center;
}

.day-buttons button {
  font-size: 3.5vw;
  width: 23vw;
  cursor: pointer;
}

input[type="date"]::-webkit-calendar-picker-indicator {
  display: block;
  cursor: pointer;
}
