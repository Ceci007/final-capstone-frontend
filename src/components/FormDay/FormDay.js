import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './FormDay.css';
import { updateDay } from '../../actions/trackings';

class FormDay extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      date: '2020-06-01',
      temperature: 0,
      selectedOption: 'üôÇ',
      medicine: ['', '', ''],
      symptons: [],
    };
  }

  componentDidMount = () => {
    const { actionToPerform, buttonId, trackings } = this.props;
    if (actionToPerform === 'Save Changes') {
      const track = trackings.filter(x => x.id.toString() === buttonId);
      this.setState({
        date: track[0].date,
        temperature: track[0].temperature,
        selectedOption: track[0].mood,
        medicine: track[0].medicines,
        symptons: track[0].symptons,
      });
    }
  }

  handleChangeDate = e => {
    this.setState({
      date: e.target.value,
    });
  }

  handleChangeTemperature = e => {
    this.setState({
      temperature: e.target.value,
    });
  }

  handleOptionChange= e => {
    this.setState({
      selectedOption: e.target.value,
    });
  }

  handleChangeMedicine = (e, index1) => {
    const { medicine } = this.state;
    medicine[index1] = e.target.value;
    this.setState({ medicine });
  }

  handleChangeSymptons = (e, index) => {
    const { symptons } = this.state;
    symptons[index] = e.target.value;
    this.setState({ symptons });
  }

  handleEdit = async (id, illnessId) => {
    const {
      date, temperature, selectedOption, medicine, symptons,
    } = this.state;
    const {
      user, updateDay, changeEditForm,
    } = this.props;
    const data = {
      id,
      userId: user.user.id,
      illnessId,
      date,
      temperature,
      mood: selectedOption,
      medicines: medicine,
      symptons,
    };

    await updateDay(data);
    changeEditForm();
  }

  handleSubmit(mood, temperature, date, medicine, symptons) {
    const { addTracking, changeAddForm, user } = this.props;
    const userId = user.user.id;
    const medicines = medicine;
    addTracking(mood, temperature, date, medicines, symptons, userId);
    changeAddForm();
  }

  render() {
    const {
      temperature, date, selectedOption, medicine, symptons,
    } = this.state;
    const {
      actionToPerform, trackings, buttonId, changeAddForm, changeEditForm,
    } = this.props;
    const track = trackings.filter(x => x.id.toString() === buttonId);

    return (
      <div>
        <h3>
          {actionToPerform}
          {' '}
          Tracking for Illness
        </h3>
        <form className="day" onSubmit={() => this.handleSubmit}>

          <div className="form-div">
            <div className="date-temp">
              <div className="date-div">
                <label htmlFor="date">
                  Date:
                  <input
                    id="date"
                    type="date"
                    name="date"
                    defaultValue={buttonId === '0' ? date : track[0].date.slice(0, 10)}
                    onChange={this.handleChangeDate}
                  />
                </label>

              </div>
              <div className="temp-div">
                <label htmlFor="temp">
                  Temperature:
                  <input
                    id="temp"
                    type="number"
                    name="temp"
                    defaultValue={buttonId === '0' ? temperature : track[0].temperature}
                    onChange={this.handleChangeTemperature}
                  />
                </label>

                <span>¬∞C</span>
              </div>
            </div>
            <div className="mood-div">
              <span>Mood: </span>
              <input type="radio" id="option1" name="mood" value="üôÇ" checked={selectedOption === 'üôÇ'} onChange={this.handleOptionChange} />
              <span role="img" aria-label="happy">üôÇ</span>
              <input type="radio" id="option2" name="mood" value="üòê" checked={selectedOption === 'üòê'} onChange={this.handleOptionChange} />
              <span role="img" aria-label="neutral">üòê</span>
              <input type="radio" id="option3" name="mood" value="üôÅ" checked={selectedOption === 'üôÅ'} onChange={this.handleOptionChange} />
              <span role="img" aria-label="sad">üôÅ</span>
              <input type="radio" id="option4" name="mood" value="üò©" checked={selectedOption === 'üò©'} onChange={this.handleOptionChange} />
              <span role="img" aria-label="sad2">üò©</span>
            </div>

            <div className="medicines2">
              <p>Medicines:</p>
              <div className="med-div">
                <input
                  id="med1"
                  type="text"
                  name="med1"
                  defaultValue={buttonId === '0' ? '' : track[0].medicines[0]}
                  placeholder="Add medicine"
                  onChange={e => this.handleChangeMedicine(e, 0, 0)}
                />
                <input
                  id="med2"
                  type="text"
                  name="med2"
                  defaultValue={buttonId === '0' ? '' : track[0].medicines[1]}
                  onChange={e => this.handleChangeMedicine(e, 1)}
                />
                <input
                  id="med3"
                  type="text"
                  name="med3"
                  defaultValue={buttonId === '0' ? '' : track[0].medicines[2]}
                  onChange={e => this.handleChangeMedicine(e, 2)}
                />
              </div>
            </div>
            <div className="symptons2">
              <p>Symptons:</p>
              <div className="med-div">
                <input
                  id="symp1"
                  type="text"
                  name="symp1"
                  placeholder="Add sympton"
                  defaultValue={buttonId === '0' ? '' : track[0].symptons[0]}
                  onChange={e => this.handleChangeSymptons(e, 0)}
                />

                <input
                  id="symp2"
                  type="text"
                  name="symp2"
                  defaultValue={buttonId === '0' ? '' : track[0].symptons[1]}
                  onChange={e => this.handleChangeSymptons(e, 1)}
                />

                <input
                  id="symp3"
                  type="text"
                  name="symp3"
                  defaultValue={buttonId === '0' ? '' : track[0].symptons[2]}
                  onChange={e => this.handleChangeSymptons(e, 2)}
                />
              </div>
            </div>
            <div className="buttons-form day-buttons">
              {actionToPerform === 'Add' && <button type="button" onClick={() => this.handleSubmit(selectedOption, temperature, date, medicine, symptons)}>{actionToPerform}</button>}
              {actionToPerform === 'Save Changes' && <button type="button" onClick={() => this.handleEdit(track[0].id, track[0].illnessId)}>Save</button>}
              {actionToPerform === 'Add' && <button type="button" onClick={changeAddForm}>Cancel</button>}
              {actionToPerform === 'Save Changes' && <button type="button" onClick={changeEditForm}>Cancel</button>}
            </div>
          </div>
        </form>
      </div>
    );
  }
}

FormDay.propTypes = {
  addTracking: PropTypes.func,
  changeEditForm: PropTypes.func,
  changeAddForm: PropTypes.func,
  updateDay: PropTypes.func,
  buttonId: PropTypes.string,
  actionToPerform: PropTypes.string,
  trackings: PropTypes.instanceOf(Array),
  user: PropTypes.shape({
    user: PropTypes.shape({
      id: PropTypes.number,
    }),
  }),
};

FormDay.defaultProps = {
  changeAddForm: () => {},
  addTracking: () => {},
  updateDay: () => {},
  changeEditForm: () => {},
  actionToPerform: '',
  trackings: [],
  buttonId: '0',
  user: {},

};

const mapStateToProps = state => ({
  user: state.user,
  trackings: state.tracking,
});
const mapDispatchToProps = dispatch => ({
  updateDay: data => dispatch(updateDay(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormDay);
