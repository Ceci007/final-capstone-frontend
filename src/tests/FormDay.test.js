import thunk from 'redux-thunk';
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combinedReducers from '../reducers';
import FormDay from '../components/FormDay/FormDay';

test('Footer renders correctly', () => {
  const store = createStore(combinedReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <FormDay />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});