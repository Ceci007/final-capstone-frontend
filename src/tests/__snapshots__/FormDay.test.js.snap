// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <h3>
    
     
    Tracking for Illness
  </h3>
  <form
    className="day"
    onSubmit={[Function]}
  >
    <div
      className="form-div"
    >
      <div
        className="date-temp"
      >
        <div
          className="date-div"
        >
          <label
            htmlFor="date"
          >
            Date:
            <input
              defaultValue="2020-06-01"
              id="date"
              name="date"
              onChange={[Function]}
              type="date"
            />
          </label>
        </div>
        <div
          className="temp-div"
        >
          <label
            htmlFor="temp"
          >
            Temperature:
            <input
              defaultValue={0}
              id="temp"
              name="temp"
              onChange={[Function]}
              type="number"
            />
          </label>
          <span>
            °C
          </span>
        </div>
      </div>
      <div
        className="mood-div"
      >
        <span>
          Mood: 
        </span>
        <input
          checked={true}
          id="option1"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="🙂"
        />
        <span
          aria-label="happy"
          role="img"
        >
          🙂
        </span>
        <input
          checked={false}
          id="option2"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="😐"
        />
        <span
          aria-label="neutral"
          role="img"
        >
          😐
        </span>
        <input
          checked={false}
          id="option3"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="🙁"
        />
        <span
          aria-label="sad"
          role="img"
        >
          🙁
        </span>
        <input
          checked={false}
          id="option4"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="😩"
        />
        <span
          aria-label="sad2"
          role="img"
        >
          😩
        </span>
      </div>
      <div
        className="medicines2"
      >
        <p>
          Medicines:
        </p>
        <div
          className="med-div"
        >
          <input
            defaultValue=""
            id="med1"
            name="med1"
            onChange={[Function]}
            placeholder="Add medicine"
            type="text"
          />
          <input
            defaultValue=""
            id="med2"
            name="med2"
            onChange={[Function]}
            type="text"
          />
          <input
            defaultValue=""
            id="med3"
            name="med3"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="symptons2"
      >
        <p>
          Symptons:
        </p>
        <div
          className="med-div"
        >
          <input
            defaultValue=""
            id="symp1"
            name="symp1"
            onChange={[Function]}
            placeholder="Add sympton"
            type="text"
          />
          <input
            defaultValue=""
            id="symp2"
            name="symp2"
            onChange={[Function]}
            type="text"
          />
          <input
            defaultValue=""
            id="symp3"
            name="symp3"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="buttons-form day-buttons"
      />
    </div>
  </form>
</div>
`;

exports[`FormDay renders correctly 1`] = `
<div>
  <h3>
    
     
    Tracking for Illness
  </h3>
  <form
    className="day"
    onSubmit={[Function]}
  >
    <div
      className="form-div"
    >
      <div
        className="date-temp"
      >
        <div
          className="date-div"
        >
          <label
            htmlFor="date"
          >
            Date:
            <input
              defaultValue="2020-06-01"
              id="date"
              name="date"
              onChange={[Function]}
              type="date"
            />
          </label>
        </div>
        <div
          className="temp-div"
        >
          <label
            htmlFor="temp"
          >
            Temperature:
            <input
              defaultValue={0}
              id="temp"
              name="temp"
              onChange={[Function]}
              type="number"
            />
          </label>
          <span>
            °C
          </span>
        </div>
      </div>
      <div
        className="mood-div"
      >
        <span>
          Mood: 
        </span>
        <input
          checked={true}
          id="option1"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="🙂"
        />
        <span
          aria-label="happy"
          role="img"
        >
          🙂
        </span>
        <input
          checked={false}
          id="option2"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="😐"
        />
        <span
          aria-label="neutral"
          role="img"
        >
          😐
        </span>
        <input
          checked={false}
          id="option3"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="🙁"
        />
        <span
          aria-label="sad"
          role="img"
        >
          🙁
        </span>
        <input
          checked={false}
          id="option4"
          name="mood"
          onChange={[Function]}
          type="radio"
          value="😩"
        />
        <span
          aria-label="sad2"
          role="img"
        >
          😩
        </span>
      </div>
      <div
        className="medicines2"
      >
        <p>
          Medicines:
        </p>
        <div
          className="med-div"
        >
          <input
            defaultValue=""
            id="med1"
            name="med1"
            onChange={[Function]}
            placeholder="Add medicine"
            type="text"
          />
          <input
            defaultValue=""
            id="med2"
            name="med2"
            onChange={[Function]}
            type="text"
          />
          <input
            defaultValue=""
            id="med3"
            name="med3"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="symptons2"
      >
        <p>
          Symptons:
        </p>
        <div
          className="med-div"
        >
          <input
            defaultValue=""
            id="symp1"
            name="symp1"
            onChange={[Function]}
            placeholder="Add sympton"
            type="text"
          />
          <input
            defaultValue=""
            id="symp2"
            name="symp2"
            onChange={[Function]}
            type="text"
          />
          <input
            defaultValue=""
            id="symp3"
            name="symp3"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="buttons-form day-buttons"
      />
    </div>
  </form>
</div>
`;
